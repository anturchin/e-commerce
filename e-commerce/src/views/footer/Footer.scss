.footer {
    position: relative;
    min-height: 12%;
    box-shadow: 0 15px 35px rgb(0, 0, 0);
    background: #222;
    color: gainsboro;
    z-index: 1000;

    display: flex;
    justify-content: space-evenly;
}

@media (max-width: 400px) {
    .footer {
        flex-direction: column;
    }
}
