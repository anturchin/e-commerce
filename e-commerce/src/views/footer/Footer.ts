import { View } from '../View';
import { Contacts } from './contacts/Contacts';
import { Address } from './address/Address';
import { Support } from './support/Support';
import './Footer.scss';

export class Footer extends View {
    constructor() {
        super({ tag: 'footer', classNames: ['footer'] });
        this.setupFooterContent();
    }

    private setupFooterContent() {
        const contacts = new Contacts().getElement();
        this.viewHtmlElement.addInnerElement(contacts);
        const address = new Address().getElement();
        this.viewHtmlElement.addInnerElement(address);
        const support = new Support().getElement();
        this.viewHtmlElement.addInnerElement(support);
    }
}
