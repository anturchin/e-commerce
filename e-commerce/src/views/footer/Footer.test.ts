import { describe, expect, test } from '@jest/globals';
import { Footer } from './Footer';

describe('Footer', () => {
    test('should match', () => {
        const footer = new Footer();
        const renderedFooter = footer.getElement();
        expect(renderedFooter.outerHTML).toMatchSnapshot();
    });
});
