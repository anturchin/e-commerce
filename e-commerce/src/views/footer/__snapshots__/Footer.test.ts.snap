// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Footer should match 1`] = `"<footer class="footer"><div class="footer__container"><img class="footer__img" src="svg-mock"><p class="subtitle">Contact</p><a class="footer__link" href="mailto:FadStore@gmail.com">E-Mail</a></div><div class="footer__container"><img class="footer__img" src="svg-mock"><p class="subtitle">Address</p><a class="footer__link" href="https://www.google.com/maps/place/%D1%83%D0%BB.+%D0%96%D1%83%D0%BA%D0%BE%D0%B2%D1%81%D0%BA%D0%BE%D0%B3%D0%BE,+%D0%A1%D0%B0%D0%BD%D0%BA%D1%82-%D0%9F%D0%B5%D1%82%D0%B5%D1%80%D0%B1%D1%83%D1%80%D0%B3,+191014/@59.9323888,30.3497675,13.57z/data=!4m15!1m8!3m7!1s0x469631a3db450ec9:0x3aad5b4a38113633!2z0YPQuy4g0JbRg9C60L7QstGB0LrQvtCz0L4sINCh0LDQvdC60YIt0J_QtdGC0LXRgNCx0YPRgNCzLCAxOTEwMTQ!3b1!8m2!3d59.935874!4d30.3558349!16s%2Fg%2F122pwsld!3m5!1s0x469631a3db450ec9:0x3aad5b4a38113633!8m2!3d59.935874!4d30.3558349!16s%2Fg%2F122pwsld?entry=ttu">Location</a></div><div class="footer__container"><img class="footer__img" src="svg-mock"><p class="subtitle">Support</p><a class="footer__link" href="#">Reference</a></div></footer>"`;
